src/App.tsx

```diff
@@ -1,223 +1,252 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodoCompleted,
+} from './api/todos';
+import { NewTodo } from './components/NewTodo';
+import { ErrorNotification } from './components/ErrorNotifications';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Filter } from './types/Filter';
+import { Notifications } from './types/Notifications';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [notification, setNotification]
+   = useState<Notifications>(Notifications.None);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [activeTodosIds, setActiveTodosIds] = useState<number[]>([]);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+  }, [title]);
+
+  const fetchTodos = async () => {
+    if (user) {
+      try {
+        setTodos(await getTodos(user.id));
+      } catch (error) {
+        setNotification(Notifications.Load);
+
+        setTimeout(() => {
+          setNotification(Notifications.None);
+        }, 3000);
```

Comment:
You are calling timeout every time you talk to API. Create one single timeout in ErrorNotifications component with UseEffect.